csharp/Svix/Models/Webhook.cs

```diff
@@ -1,116 +1,117 @@
-                msgTimestamp = headers.Get(UNBRANDED_TIMESTAMP_HEADER_KEY);
-                if (String.IsNullOrEmpty(msgId) || String.IsNullOrEmpty(msgSignature) || String.IsNullOrEmpty(msgTimestamp))
-                {
-                    throw new WebhookVerificationException("Missing Required Headers");
-                }
-            }
-
-            var timestamp = Webhook.VerifyTimestamp(msgTimestamp);
-
-            var signature = this.Sign(msgId, timestamp, payload);
-            var expectedSignature = signature.Split(",")[1];
-
-            var passedSignatures = msgSignature.Split(' ');
-            foreach (string versionedSignature in passedSignatures)
-            {
-                var parts = versionedSignature.Split(",");
-                if (parts.Length < 2)
-                {
-                    throw new WebhookVerificationException("Invalid Signature Headers");
-                }
-                var version = parts[0];
-                var passedSignature = parts[1];
-
-                if (version != "v1")
-                {
-                    continue;
-                }
-                if (Utils.SecureCompare(expectedSignature, passedSignature))
-                {
-                    return;
-                }
-
-            }
-            throw new WebhookVerificationException("No matching signature found");
-        }
-
-        private static DateTimeOffset VerifyTimestamp(string timestampHeader)
-        {
-            DateTimeOffset timestamp;
-            var now = DateTimeOffset.UtcNow;
-            try
-            {
-                var timestampInt = long.Parse(timestampHeader);
-                timestamp = DateTimeOffset.FromUnixTimeSeconds(timestampInt);
-            }
-            catch
-            {
-                throw new WebhookVerificationException("Invalid Signature Headers");
-            }
-
-            if (timestamp < (now.AddSeconds(-1 * TOLERANCE_IN_SECONDS)))
-            {
-                throw new WebhookVerificationException("Message timestamp too old");
-            }
-            if (timestamp > (now.AddSeconds(TOLERANCE_IN_SECONDS)))
-            {
-                throw new WebhookVerificationException("Message timestamp too new");
-            }
-            return timestamp;
-        }
-
-        public string Sign(string msgId, DateTimeOffset timestamp, string payload)
-        {
-            var toSign = $"{msgId}.{timestamp.ToUnixTimeSeconds().ToString()}.{payload}";
-            var toSignBytes = SafeUTF8Encoding.GetBytes(toSign);
-            using (var hmac = new HMACSHA256(this.key))
-            {
-                var hash = hmac.ComputeHash(toSignBytes);
-                var signature = Convert.ToBase64String(hash);
-                return $"v1,{signature}";
-            }
-        }
-    }
+﻿using System;
+using System.Text;
+using System.Net;
+using System.Security.Cryptography;
+
+using Svix.Exceptions;
+
+namespace Svix.Models
```

Comment:
Unfortunately moving the Webhook class to the `Models` namespace is a breaking change, I think its best we leave this in the `Svix` namespace for now.

Reply:
I agree. I think moving this back to the Svix namespace makes more sense given your later comment about the Api client.